### PR TITLE
[timer] Add gauges to csharp timers to reflect python gauges.

### PR DESCRIPTION
**Pull request by surfnerd**

### CODE REVIEW

File: UnitySDK/Assets/ML-Agents/Scripts/Timer.cs [hunk 1]
```diff
@@ -250,6 +294,7 @@ public void Reset(string name = "root")
         {
             m_Stack = new Stack<TimerNode>();
             m_RootNode = new TimerNode(name, true);
+            m_RootNode.Gauges = new Dictionary<string, GaugeNode>();
```
Comment (chriselion) [inline UnitySDK/Assets/ML-Agents/Scripts/Timer.cs]: nit: maybe do this in TimerNode.TimerNode if `isRoot` is true? Could also create lazily or store in TimerStack instead, if you're worried about memory.

File: UnitySDK/Assets/ML-Agents/Scripts/Timer.cs [hunk 2]
```diff
@@ -75,6 +82,12 @@ public double TotalSeconds
             set {}  // Serialization needs this, but unused.
         }
 
+        public Dictionary<string, GaugeNode> Gauges
+        {
+            get { return m_Gauges; }
+            set { m_Gauges = value; }
```
Comment (chriselion) [inline UnitySDK/Assets/ML-Agents/Scripts/Timer.cs]: I left the setter empty in some of the other properties; unfortunately serialization seems to need them, but we don't really want them settable, and Rider complains about the unused `value` parameter if they're unimplemented.

I don't really care either way, but we probably should be consistent.

Comment (surfnerd) [reply UnitySDK/Assets/ML-Agents/Scripts/Timer.cs]: took your suggestion.

File: UnitySDK/Assets/ML-Agents/Scripts/Timer.cs (hunk 1 above)
Comment (surfnerd) [reply UnitySDK/Assets/ML-Agents/Scripts/Timer.cs]: took your suggestion

**Pull request merged**